### PR TITLE
[14.0] [IMP] cetmix_tower_server: UI improvements

### PR DESCRIPTION
**Pull request by Aldeigja**
list of UI improvements introduced in this PR:

- Command form: hide the "Code" tab when "Upload file" and "Execute Plan" actions are selected.
- Command form: add placeholder to the "Code" field. Need to avoid misusage.
- Command form: add archive action.
- Server form: move "Test connection" button to the general settings tab. Make it available for Manager and Root only.
- Server form: add "Run Command" and "Run Flight Plan" buttons in the form header.
- Server form:  hide configuration and secrets tab on the server form  from group_user.
- Flight plan form: update command access warning message.
- Flight plan form: add archive action.
- System: change "Execute" to  "Run" on buttons, menus and labels. Python method names remain the same.
- System: add tower.server.os​ system variable that returns OS name.

Task: 4083

## Release Notes

- **New Features**
  - Introduced a new SSH command template for improved command execution.
  - Added buttons for executing commands and flight plans in the server management interface.
  - Enhanced search functionality with filters for archived commands and plans.

- **User Interface Improvements**
  - Added visual indicators for archived commands and plans.
  - Adjusted field visibility and requirements in forms for better user experience.
  - Updated button and form titles for consistency and clarity.

- **Bug Fixes**
  - Improved clarity of warning messages related to command access levels.

- **Enhanced Information**
  - Included operating system information in server variable outputs.

These updates enhance functionality and usability across various components of the system.

### CODE REVIEW

File: cetmix_tower_server/models/cx_tower_command.py [hunk 1]
```diff
@@ -31,6 +31,11 @@
 # }
 \n\n\n"""
 
+DEFAULT_SSH_CODE = """ #Execute any SSH command on the target system
+# Examples: ls, cd, pwd, mkdir, rm
+# Adapt commands to your specific OS.
+\n\n\n"""
```
[the remark this answers is not in the document]
Comment (Aldeigja) [reply cetmix_tower_server/models/cx_tower_command.py]: I'm not sure we really need these guidelines here.

File: cetmix_tower_server/models/cx_tower_command.py [hunk 2]
```diff
@@ -31,6 +31,11 @@
 # }
 \n\n\n"""
 
+DEFAULT_SSH_CODE = """ #Execute any SSH command on the target system
```
Comment (ivs-cetmix) [inline cetmix_tower_server/models/cx_tower_command.py]: `"""# Execute a shell command using SSH on the remote system`
image

File: cetmix_tower_server/views/cx_tower_server_view.xml [hunk 3]
```diff
@@ -150,10 +150,23 @@
...
+                    <button
+                        name="action_execute_command"
+                        type="object"
+                        string="Execute command"
+                        icon="fa-code"
+                    />
+                <button
+                        name="action_execute_plan"
+                        type="object"
+                        string="Execute Flight plan"
```
Comment (ivs-cetmix) [inline cetmix_tower_server/views/cx_tower_server_view.xml]: `Execute Flight Plan`

**Review by ivs-cetmix — CHANGES_REQUESTED** [review]

File: cetmix_tower_server/views/cx_tower_server_view.xml [hunk 4]
```diff
@@ -205,17 +223,31 @@
...
-                                    <field name="ip_v4_address" />
-                                    <field name="ip_v6_address" />
+                                    <field
+                                        name="ip_v4_address"
+                                        groups="cetmix_tower_server.group_manager"
+                                    />
+                                    <field
+                                        name="ip_v6_address"
+                                        groups="cetmix_tower_server.group_manager"
+                                    />
```
[the remark this answers is not in the document]
Comment (Aldeigja) [reply cetmix_tower_server/views/cx_tower_server_view.xml]: Members of the group_user don't have access to the server templates and creation of templates and servers. Your comment is not applicable here.

File: cetmix_tower_server/models/cx_tower_command.py [hunk 5]
```diff
@@ -31,6 +31,11 @@
 # }
 \n\n\n"""
 
+DEFAULT_SSH_CODE = """ # Execute any SSH command on the target system
```
Comment (ivs-cetmix) [inline cetmix_tower_server/models/cx_tower_command.py]: Please remove the trailing whitespace
`""" # Execute any SSH command on the target system`
->
`"""# Execute any SSH command on the target system`

File: cetmix_tower_server/models/cx_tower_plan.py [hunk 6]
```diff
@@ -260,9 +260,9 @@ def _compute_command_access_level(self):
                 command_names = ", ".join(commands_with_higher_access.mapped("name"))
                 record.access_level_warn_msg = _(
                     "The access level of command(s) '%(command_names)s' included in the"
-                    "current Flight plan is higher than  the access level of the Flight"
-                    "plan itself. Ensure that you really want to give access to those"
-                    "command(s) to the flight plan users.",
+                    " current Flight plan is higher than the access level of the"
+                    " Flight plan itself. Ensure that you really want to give access"
+                    " to these command(s) to the Flight plan users.",
```
Comment (ivs-cetmix) [inline cetmix_tower_server/models/cx_tower_plan.py]: Please ensure that you want to allow those commands to be tun anyway.

File: cetmix_tower_server/views/cx_tower_server_view.xml [hunk 7]
```diff
@@ -230,10 +262,26 @@
                                         attrs="{'required': [('ssh_auth_mode', '=', 'k')]}"
                                     />
                                 </group>
-                                <field name="note" placeholder="Put your notes here" />
+                                <group name="test_connection_group">
```
Comment (ivs-cetmix) [inline cetmix_tower_server/views/cx_tower_server_view.xml]: This button should be available for Manager and Root groups only.
Also we should move it closer to the IP Address Fields
image

**Review by ivs-cetmix — CHANGES_REQUESTED** [review]

**Comment by CetmixGitDrone** [comment]
Congratulations, PR rebased to [14.0-dev](https://github.com/cetmix/cetmix-tower/commits/14.0-dev).

File: cetmix_tower_server/models/cx_tower_command.py [hunk 8]
```diff
@@ -31,6 +31,11 @@
 # }
 \n\n\n"""
 
+DEFAULT_SSH_CODE = """# Execute any SSH command on the target system
```
Comment (ivs-cetmix) [inline cetmix_tower_server/models/cx_tower_command.py]: Execute -> Run

**Review by ivs-cetmix — CHANGES_REQUESTED** [review]

File: cetmix_tower_server/views/cx_tower_server_view.xml [hunk 9]
```diff
@@ -145,15 +145,24 @@
...
+                        string="Run command"
+                        icon="fa-code"
+                    />
+                <button
+                        name="action_execute_plan"
                         type="object"
-                        string="Test Connection"
+                        string="Run Flight Plan"
+                        icon="fa-plane"
                     />
```
[the remark this answers is not in the document]
Comment (Aldeigja) [reply cetmix_tower_server/views/cx_tower_server_view.xml]: > * Add group restrictions to limit access to Manager and Root groups

These buttons don't need any restrictions, they should be accessible for members of group user.

> * Move the buttons closer to the IP Address fields for better UX

In my opinion, the form header is the most proper place for these buttons

**Review by ivs-cetmix — APPROVED** [review]
LGTM

File: cetmix_tower_server/views/cx_tower_server_view.xml [hunk 10]
```diff
@@ -145,15 +145,24 @@
...
                 <header>
+
                     <button
-                        name="test_ssh_connection"
+                        name="action_execute_command"
+                        type="object"
+                        string="Run command"
+                        icon="fa-code"
+                    />
+                <button
```
Comment (tendil) [inline cetmix_tower_server/views/cx_tower_server_view.xml]: Add tab to the 155 line for button

---

Снимок экрана 2024-12-02 в 13 34 14

File: cetmix_tower_server/views/cx_tower_server_view.xml [hunk 11]
```diff
@@ -214,7 +223,17 @@
                                     />
                                     <field name="ip_v4_address" />
                                     <field name="ip_v6_address" />
+                                    <br />
+                                <button
```
Comment (tendil) [inline cetmix_tower_server/views/cx_tower_server_view.xml]: Add tab to the 227 line for button

---

Снимок экрана 2024-12-02 в 13 37 49

**Review by tendil — COMMENTED** [review]
Check Review

**Review by tendil — APPROVED** [review]
LGTM

**Comment by CetmixGitDrone** [comment]
This PR has the `approved` label and has been created more than 5 days ago. It should therefore be ready to merge by a maintainer (or a PSC member if the concerned addon has no declared maintainer). 🤖

**Comment by CetmixGitDrone** [comment]
What a great day to merge this nice PR. Let's do it!
Prepared branch [14.0-dev-ocabot-merge-pr-149-by-ivs-cetmix-bump-nobump](https://github.com/cetmix/cetmix-tower/commits/14.0-dev-ocabot-merge-pr-149-by-ivs-cetmix-bump-nobump), awaiting test results.

**Comment by CetmixGitDrone** [comment]
Congratulations, your PR was merged at a95d056e9eed42e988a5693dfc3c7be886917692. Thanks a lot for contributing to cetmix. ❤️

**Pull request merged**